.github/workflows/sync-translations-poeditor.yml

```diff
@@ -0,0 +1,200 @@
+name: Sync Translations with POEditor
+
+on:
+  push:
+    branches:
+      - feat/translation-poeditor-integration
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  sync-translations:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Configure Git
+        run: |
+          git config user.name ${{ secrets.GIT_BOT_USERNAME }}
+          git config user.email ${{ secrets.GIT_BOT_EMAIL }}
+          git remote set-url origin https://${{ secrets.GIT_BOT_ACCESS_TOKEN }}@github.com/Aam-Digital/ndb-core.git
+
+      - name: Create or update branch for translations
+        run: |
+          branch_name="chore/update-poeditor-translations"
+
+          # Check if branch exists on remote
+          if git ls-remote --exit-code origin $branch_name; then
+            echo "Branch $branch_name already exists. Fetching it..."
+            git fetch origin $branch_name:$branch_name
+
+            git checkout $branch_name
+            git pull origin $branch_name
+          else
+            echo "Creating new branch $branch_name..."
+            git checkout -b $branch_name
+          fi
+
+      - name: Extract i18n keys
+        run: npm run extract-i18n
+
+      - name: Check if any files changed
+        run: |
+          if git diff --quiet; then
+            echo "No changes detected."
+            echo "changed=false" >> $GITHUB_ENV
+          else
+            echo "Changes detected."
+            echo "changed=true" >> $GITHUB_ENV
+          fi
```

Comment:
@tomwwinter Yes, this can be managed at the GitHub Action level, and I have implemented it that way before. However, following Sebastian's suggestion, we can handle this through CI. For reference, please check this [comments](https://github.com/Aam-Digital/ndb-core/pull/2672#discussion_r1856692659). Please let me know, if I need to changes this.

Reply:
Thanks, missed that conversation.